### PR TITLE
fix: gRPC Version

### PR DESCRIPTION
**Pull request by MatKuhr**
## Context

The approach of only using `dependencyManagement` doesn't work, as this only affects the version within the module itself, not the version when using the module.

Unfortunately, we can only override the version by referencing transitive dependencies explicitly. We could add it to the main BOM, but that only fixes it when the BOM is used by customers. 

## Definition of Done

- [x] Functionality scope stated & covered
- [x] [Tests cover the scope above](https://sdk-v2.cpe.c.eu-de-1.cloud.sap/job/end-to-end-tests/view/Cloud%20SDK%20v5/job/v5-scp-cf-spring-ias/465/)
- [x] ~Error handling created / updated & covered by the tests above~
- [x] ~Documentation updated~
- [x] Release notes updated

### CODE REVIEW

File: cloudplatform/connectivity-ztis/pom.xml [hunk 1]
```diff
@@ -87,6 +87,22 @@
 			<version>0.8.6</version>
```
Comment (CharlesDuboisSAP) [inline cloudplatform/connectivity-ztis/pom.xml:87]: `java-spiffe-core` also uses `grpc`
image

File: cloudplatform/connectivity-ztis/pom.xml [hunk 2]
```diff
@@ -87,6 +87,22 @@
...
+		<dependency>
+			<groupId>io.grpc</groupId>
+			<artifactId>grpc-protobuf</artifactId>
+			<scope>runtime</scope>
+		</dependency>
+		<dependency>
+			<groupId>io.grpc</groupId>
+			<artifactId>grpc-stub</artifactId>
+			<scope>runtime</scope>
+		</dependency>
```
Comment (CharlesDuboisSAP) [inline cloudplatform/connectivity-ztis/pom.xml:105]: I thought only `grpc-netty-shaded` would need to be excluded, why also `grpc-protobuf` and `grpc-stub`?
image

File: cloudplatform/connectivity-ztis/pom.xml (hunk 1 above)
Comment (MatKuhr) [reply cloudplatform/connectivity-ztis/pom.xml:87]: the versions 1.62.0 will be omitted by maven in favor of 1.66.0

File: cloudplatform/connectivity-ztis/pom.xml (hunk 2 above)
Comment (MatKuhr) [reply cloudplatform/connectivity-ztis/pom.xml:105]: no need to exclude, we only need to force the version

**Pull request merged**